web/src/main/java/com/microsoft/applicationinsights/web/extensibility/modules/SpringMvcRequestTrackingTelemetryModule.java

```diff
@@ -0,0 +1,74 @@
+package com.microsoft.applicationinsights.web.extensibility.modules;
+
+import com.microsoft.applicationinsights.TelemetryClient;
+import com.microsoft.applicationinsights.TelemetryConfiguration;
+import com.microsoft.applicationinsights.extensibility.TelemetryModule;
+import com.microsoft.applicationinsights.internal.logger.InternalLogger;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.web.servlet.DispatcherServlet;
+
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.Arrays.asList;
+
+public class SpringMvcRequestTrackingTelemetryModule implements WebTelemetryModule, TelemetryModule {
+
+    private TelemetryClient telemetryClient;
+    private List<String> exceptionAttributesList = asList(DispatcherServlet.class.getName() + ".EXCEPTION");
+    private String exceptionAttributes;
+
+    @Override
+    public void initialize(TelemetryConfiguration configuration) {
+        telemetryClient = new TelemetryClient(configuration);
+    }
+
+    @Override
+    public void onBeginRequest(ServletRequest req, ServletResponse res) {
+
+    }
+
+    @Override
+    public void onEndRequest(ServletRequest req, ServletResponse res) {
```

Comment:
I believe this is invoked through `safeInvokeOnEndRequest` (or something like it) which wraps the invocation with try/catch